OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/reflection/ReflectionRelatedCallsAnalysis.scala

```diff
@@ -712,112 +741,124 @@ class MethodInvokeAnalysis private[analyses] (
+            implicit val highSoundness = HighSoundnessMode("class")
+            val (callPC, receiver, params, matchers, _, _) = state.dependersOf(epk).head.asInstanceOf[classDependerType]
 
-                if (nameMatcher ne NoMethodsMatcher) {
-                    val matchers = data._4 + nameMatcher
-                    val allMatchers = matchers +
-                        MatcherUtil.retrieveClassBasedMethodMatcher(
-                            data._8,
-                            data._7,
-                            (data._1, data._2, data._3, matchers, data._7, data._6),
-                            data._1,
-                            stmts,
-                            project,
-                            () => failure(data._1, data._2, data._3, matchers),
-                            onlyMethodsExactlyInClass = !data._4.contains(PublicMethodMatcher)
-                        )
+            val classes = eps.asInstanceOf[EPS[_, ForNameClasses]].ub.classes.map { tpe =>
+                if (tpe.isObjectType) tpe.asObjectType else ObjectType.Object
+            }
+
+            val allMatchers = matchers + retrieveSuitableMatcher[Set[ObjectType]](
+                Some(classes),
+                callPC,
+                v => new ClassBasedMethodMatcher(v, !matchers.contains(PublicMethodMatcher))
+            )
 
+            addCalls(state.callContext, callPC, _ => receiver, params, allMatchers)
+        } else {
+            implicit val highSoundness = HighSoundnessMode("method")
+
+            AllocationsUtil.continuationForAllocation[methodDependerType, ContextType](
+                eps, state.callContext, _ => (methodVar, state.tac.stmts),
+                _.isInstanceOf[(_, _, _, _)], data => failure("method", data._1, data._2, data._3, data._4)
+            ) { (data, allocationContext, allocationIndex, stmts) =>
+                    val allMatchers = handleGetMethod(
+                        allocationContext, data._1, allocationIndex, data._2, data._3, data._4, stmts
+                    )
                     addCalls(state.callContext, data._1, _ => data._2, data._3, allMatchers)
                 }
-            }
 
-        AllocationsUtil.continuationForAllocation[classDependerType, ContextType](
-            eps, state.callContext, data => (data._5, data._6),
-            _.isInstanceOf[(_, _, _, _, _, _)], data => failure(data._1, data._2, data._3, data._4)
-        ) { (data, allocationContext, allocationIndex, stmts) =>
-                val classes = TypesUtil.getPossibleClasses(
-                    allocationContext, allocationIndex, data,
-                    stmts, project, () => failure(data._1, data._2, data._3, data._4),
-                    onlyObjectTypes = false
-                )
+            AllocationsUtil.continuationForAllocation[nameDependerType, ContextType](
+                eps, state.callContext, data => (data._5, data._6),
+                _.isInstanceOf[(_, _, _, _, _, _, _, _)], data => {
```

Comment:
Can't you use TupleX here? And maybe match data, so one knows what actually is in there.

Reply:
This will hopefully be done as a side-task of #116